### ISSUE TITLE
PreAuthorization attrs nil or false

### ISSUE THREAD

**Issue by Nilpointer** [doorkeeper-gem/doorkeeper-openid_connect#140]
Hi all,

sorry I am new to this project and topic. I set up doorkeeper and dk-openidconnect.
When trying to authorize I run in to an error in the initialize method at

```
module PreAuthorization
        attr_reader :nonce

        def initialize(server, client, attrs = {})
          super
          @nonce = attrs[:nonce]
        end
      end
```

The attrs property is nil if I use the doorkeeper config as recommended:
```
resource_owner_authenticator do
    if current_user
      current_user
    else
      redirect_to(new_user_session_url)
      nil
    end
  end
```

The documentation says the resource_owner_authenticator should return something "falsy" if nor user logged in, but I only get this to work when returning an empty hash instead of nil (or false).

Is returning an empty hash ok or am I breaking things helping me this way?

Regards
Andreas

**Comment by teslat** [comment]
I was have the same issue with Doorkeeper 5.5 but it resolved when I downgrade to Doorkeeper 5.4

**Comment by toupeira** [comment]
Thanks for the report! :+1: 

It looks like you're using an old version of `doorkeeper-openid_connect`, the signature for `PreAuthorization#initialize` was changed from `def initialize(server, client, attrs = {})` for 1.7.0 in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/commit/3f53e0f.

If you were still experiencing this with 1.7.5, please also try 1.8.0.pre.rc1 which was just released (and also requires Doorkeeper 5.5 / Ruby 2.5 now).
